### ISSUE TITLE
Don't use `DvcsUtil.guessRepositoryForFile` without a read action

### ISSUE THREAD

**Issue by vlad20012** [Almighty-Alpaca/JetBrains-Discord-Integration#109]
Hello!
I'm from JetBrains. Recently we started to receive error reports related to this plugin. The first report came on 08.06.2020.

<details>
  <summary>Exception sample</summary>
  
  ```

java.lang.IllegalStateException: Needs read action
	at org.rust.openapiext.UtilsKt.checkReadAccessAllowed(utils.kt:80)
	at org.rust.cargo.project.workspace.RsAdditionalLibraryRootsProvider.getAdditionalProjectLibraries(RsAdditionalLibraryRootsProvider.kt:53)
	at com.intellij.openapi.roots.impl.RootIndex.buildRootInfo(RootIndex.java:238)
	at com.intellij.openapi.roots.impl.RootIndex.getOrderEntryGraph(RootIndex.java:320)
	at com.intellij.openapi.roots.impl.RootIndex.getOrderEntries(RootIndex.java:965)
	at com.intellij.openapi.roots.impl.DirectoryIndexImpl.getOrderEntries(DirectoryIndexImpl.java:180)
	at com.intellij.openapi.roots.impl.ProjectFileIndexImpl.getOrderEntriesForFile(ProjectFileIndexImpl.java:121)
	at com.intellij.dvcs.DvcsUtil.getVcsRootForLibraryFile(DvcsUtil.java:367)
	at com.intellij.dvcs.DvcsUtil.guessVcsRoot(DvcsUtil.java:404)
	at com.intellij.dvcs.DvcsUtil.guessRepositoryForFile(DvcsUtil.java:286)
	at com.almightyalpaca.jetbrains.plugins.discord.plugin.extensions.git.GitVcsInfoExtension.getCurrentVcsBranch(GitVcsInfoExtension.kt:12)
	at com.almightyalpaca.jetbrains.plugins.discord.plugin.extensions.VcsInfoExtension$Companion$getCurrentVcsBranch$1.invoke(VcsInfoExtension.kt:18)
	at com.almightyalpaca.jetbrains.plugins.discord.plugin.extensions.VcsInfoExtension$Companion$getCurrentVcsBranch$1.invoke(VcsInfoExtension.kt:11)
	at kotlin.sequences.TransformingSequence$iterator$1.next(Sequences.kt:172)
	at kotlin.sequences.FilteringSequence$iterator$1.calcNext(Sequences.kt:132)
	at kotlin.sequences.FilteringSequence$iterator$1.hasNext(Sequences.kt:156)
	at kotlin.sequences.SequencesKt___SequencesKt.firstOrNull(_Sequences.kt:129)
	at com.almightyalpaca.jetbrains.plugins.discord.plugin.extensions.VcsInfoExtension$Companion.getCurrentVcsBranch(VcsInfoExtension.kt:20)
	at com.almightyalpaca.jetbrains.plugins.discord.plugin.data.DataService.getDataInternal(DataService.kt:115)
	at com.almightyalpaca.jetbrains.plugins.discord.plugin.data.DataService.getData(DataService.kt:49)
	at com.almightyalpaca.jetbrains.plugins.discord.plugin.render.RenderService$render$2.invokeSuspend(RenderService.kt:63)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:56)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:738)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665)

  ```
  
</details>

As far as I know, it is not correct to invoke `DvcsUtil.guessRepositoryForFile` without a read action. Unfortunately, it isn't documented anywhere :(

Also, the issue may look related to Rust plugin, but really it isn't - Rust plugin just asserts that there is a read action.

**Comment by Almighty-Alpaca** [comment]
The Issue report you received was actually from a contributor of my plugin. We assumed since it's not documented the Rust plugin wrongly asserts that there is a ReadAction.
Thank you for the clarification! I will fix this in this plugin then.

**Comment by Almighty-Alpaca** [comment]
A fix for this has just been released with v1.5.0.
